### DISCUSSION TITLE
is ther any way to add our own mutations to serving graphql server

### DISCUSSION THREAD

**Discussion by ethdev279** [graphprotocol/graph-client#207] [Q&A] [answered]
When I run `graphclient serve-dev`, I can see graphql server wrapping existing subgraph api.  I'm wondering if we can extend our graph client server with additional custom mutations/queries?
@ardatan

**Comment by ardatan** [comment, marked as answer]
Yes! You can extend the schema with `additionalTypeDefs` and `additionalResolvers`;
https://github.com/graphprotocol/graph-client#client-side-mutations

**Reply by ethdev279** [reply]
Exactly what I needed. Thanks much!

**Reply by ethdev279** [reply]
getting this now: `Schema is not configured to execute mutation operation.` can you pls help me? @ardatan

**Reply by ardatan** [reply]
Could you create a reproduction on CodeSandbox or StackBlitz? So we can move better.

**Reply by ardatan** [reply]
I think It is missing in the docs. You have add this on top of `additionalTypedefs` ;
```graphql
extend schema {
   mutation: Mutation
}
```

I created this PR to update docs.
https://github.com/graphprotocol/graph-client/pull/208
